src/helpers.ts

```diff
@@ -162,3 +164,20 @@ export function toIterable<T>(i: any): any {
     }
     return [i]; // a simple value
 }
+
+/**
+ * Merge multiple iterators or iterables in to a single iterable.
+ * Merged inputs are iterated over in the order in which they were specified.
+ *
+ * Note that if you concatenate asynchronous iterables with a synchronous pipeline, they will be processed as simple values.
+ *
+ * @see [[concat]]
+ * @category Sync+Async
+ */
+export function mergeIterables<Ts extends [...unknown[]]>(
+    ...iters: {
+        [P in keyof Ts]: Any<Ts[P]>;
+    }
+) {
+    return pipe([], concat<never, Ts>(...iters));
```

Comment:
This relies on #110 new type info. CI won't pass until this PR is rebase off that one.